MC/Core/Src/freertos.c

```diff
@@ -0,0 +1,390 @@
+void MC_SendStartThread(void *argument);
+void MC_SendEndThread(void *argument);
+
+/* USER CODE END FunctionPrototypes */
+
+void StartDefaultTask(void *argument);
+
+void MX_FREERTOS_Init(void); /* (MISRA C 2004 rule 8.1) */
+
+/**
+  * @brief  FreeRTOS initialization
+  * @param  None
+  * @retval None
+  */
+void MX_FREERTOS_Init(void) {
+  /* USER CODE BEGIN Init */
+
+  /* USER CODE END Init */
+
+  /* USER CODE BEGIN RTOS_MUTEX */
+  /* add mutexes, ... */
+  /* USER CODE END RTOS_MUTEX */
+
+  /* USER CODE BEGIN RTOS_SEMAPHORES */
+
+  MCSemStartHandle = osSemaphoreNew(1, 0, &MCSemStart_attributes);
+  MCSemEndHandle = osSemaphoreNew(1, 0, &MCSemEnd_attributes);
+  /* add semaphores, ... */
+  /* USER CODE END RTOS_SEMAPHORES */
+
+  /* USER CODE BEGIN RTOS_TIMERS */
+  /* start timers, add new ones, ... */
+  /* USER CODE END RTOS_TIMERS */
+
+  /* USER CODE BEGIN RTOS_QUEUES */
+  start_frame_q = osMessageQueueNew(1, sizeof(CAN_Frame_t), NULL);
+  end_frame_q = osMessageQueueNew(1, sizeof(CAN_Frame_t), NULL);
+
+  /* add queues, ... */
+  /* USER CODE END RTOS_QUEUES */
+
+  /* Create the thread(s) */
+  /* creation of defaultTask */
+
+  /* USER CODE BEGIN RTOS_THREADS */
+
+  MC_ReceiveStartTaskHandle = osThreadNew(MC_ReceiveStartThread, NULL, MC_ReceiveStartTaskAttr);
+  MC_ReceiveEndTaskHandle = osThreadNew(MC_ReceiveEndThread, NULL, MC_ReceiveEndTaskAttr);
+  MC_SendStartTaskHandle = osThreadNew(MC_SendStartThread, NULL, MC_SendStartTaskAttr);
+  MC_SendEndTaskHandle = osThreadNew(MC_SendEndThread, NULL, MC_SendEndTaskAttr);
+
+  /* add threads, ... */
+  /* USER CODE END RTOS_THREADS */
+
+  /* USER CODE BEGIN RTOS_EVENTS */
+  /* add events, ... */
+  /* USER CODE END RTOS_EVENTS */
+
+}
+
+/* USER CODE BEGIN Header_StartDefaultTask */
+/**
+  * @brief  Function implementing the defaultTask thread.
+  * @param  argument: Not used
+  * @retval None
+  */
+/* USER CODE END Header_StartDefaultTask */
+void StartDefaultTask(void *argument)
+{
+  /* USER CODE BEGIN StartDefaultTask */
+  /* Infinite loop */
+  for(;;)
+  {
+    osDelay(1);
+  }
+  /* USER CODE END StartDefaultTask */
+}
+
+/* Private application code --------------------------------------------------*/
+/* USER CODE BEGIN Application */
+
+void MC_ReceiveStartThread(void *argument) {
+
+	CAN_Frame_t start_frame = CAN_frame_init(&hcan3, MOTOR_CONTROLLER_K1); //Initialize CAN_Frame to send to MC to receive first data frame
+
+	start_frame.id_type = CAN_ID_EXT;
+	start_frame.data_length = 0;
+	start_frame.rtr = 1; //Configure the request transmission bit to high
+
+	uint8_t received = 0;
+
+	for(;;){
+
+
+		// Checks if mailbox is available for transmission and sends start_frame to the MC (requesting data bacK)
+	    if (HAL_CAN_GetTxMailboxesFreeLevel(&hcan3)) {
+			CAN_send_frame(start_frame);
+	    }
+
+		if(HAL_CAN_GetRxFifoFillLevel(&hcan3, CAN_RX_FIFO0)){
+			start_frame = CAN_get_frame(&hcan3, CAN_RX_FIFO0);
+			received = 1;
```

Comment:
why not just a boolean

Reply:
Have to import stdbool, 8 bit integer works fine in this case and uses less memory

Reply:
As in less memory than importing stdbool? Fair enough, either works